File: 703-kth-largest-element-in-a-stream/703-kth-largest-element-in-a-stream.py
class KthLargest:
    def __init__(self, k: int, nums: List[int]):
        self.nlargest = k;
        self.stream = nums
        

    def add(self, val: int) -> int:
        # print(self.nlargest, self.stream)
        self.stream.append(val)
        self.stream.sort(reverse=True)
        # print(self.stream, self.nlargest, self.stream[self.nlargest-1])
        return self.stream[self.nlargest - 1]


# Your KthLargest object will be instantiated and called as such:
# obj = KthLargest(k, nums)
# param_1 = obj.add(val)

File: 703-kth-largest-element-in-a-stream/703-kth-largest-element-in-a-stream.js
/**
 * @param {number} k
 * @param {number[]} nums
 */
var KthLargest = function(k, nums) {
    this.k = k
    this.stream = nums;
    
};

/** 
 * @param {number} val
 * @return {number}
 */
KthLargest.prototype.add = function(val) {
    this.stream.push(val);
    this.stream = this.stream.sort((a,b) => b-a)
   //console.log(this.stream, this.k)
    return this.stream[this.k-1];
};

/** 
 * Your KthLargest object will be instantiated and called as such:
 * var obj = new KthLargest(k, nums)
 * var param_1 = obj.add(val)
 */